tools/ubuntu-setup.sh

```diff
@@ -0,0 +1,272 @@
+
+# Copyright 2018 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+# ubuntu-setup.sh setups a syzkaller fuzz env automatically.
+
+# 1. C compiler with coverage support
+# 2. Linux kernel with coverage additions
+# 3. Virtual machine or a physical device
+# 4. Syzkaller fuzz
+
+# Author: Nixawk
+# Date  : 2018/12/08
+# Lab   : Ubuntu 18.04 x86_64 (4.19.6)
+
+export INIT_DIR=`pwd`
+export KERNEL_RELEASE_URL="https://www.kernel.org/releases.json"
+export KERNEL_RELEASE_VERSION=$(curl -s $KERNEL_RELEASE_URL | grep -A2 "latest_stable" | sed ':a;N;$!ba;s/\n//g' | grep -oP '\d.\d+.\d')
+export KERNEL_SOURCE_URL="https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-$KERNEL_RELEASE_VERSION.tar.xz"
+export KERNEL_PATCH_URL="https://cdn.kernel.org/pub/linux/kernel/v4.x/patch-$KERNEL_RELEASE_VERSION.xz"
+export GOLANG_URL="https://dl.google.com/go/go1.11.2.linux-amd64.tar.gz"
+export KERNEL_SOURCE_FILE=$(basename $KERNEL_SOURCE_URL)
+export KERNEL_PATCH_FILE=$(basename $KERNEL_PATCH_URL)
+export GOLANG_FILE=$(basename $GOLANG_URL)
+export GOROOT="$INIT_DIR/go"
+export GOPATH="$INIT_DIR/gopath"
+export PATH=$GOROOT/bin:$PATH
+export KERNEL_SOURCE_DIR="${KERNEL_SOURCE_FILE%.*.*}"
+export KBLD_DIR="$INIT_DIR/build_kernel"
+export KCGF_FILE="$KBLD_DIR/.config"
+export IMG_RELEASE="stretch"  # /usr/share/debootstrap/scripts/stretch
+export IMG_BLD_DIR="$INIT_DIR/build_img"
+export IMG_FILE="$INIT_DIR/$IMG_RELEASE.img"
+
+echo "[*] Syzkaller is trying to use [linux-$KERNEL_RELEASE_VERSION]"
+
+function install_dependencies {
+    echo "[*] install all dependencies"
+    sudo apt-get install -qq -y flex bison libc6-dev libc6-dev-i386 \
+	    linux-libc-dev linux-libc-dev:i386 libgmp3-dev \
+	    libmpfr-dev libmpc-dev build-essential bc \
+            debootstrap qemu-system-x86
+}
+
+function check_gcc_coverage {
+    echo "[*] check if gcc has coverage feature"
+    OPT_COVERAGE=$(MANWIDTH=160 man gcc | grep '      \-\-coverage')
+    if [[ "$OPT_COVERAGE" == *"--coverage"* ]]; then
+	echo "[*] gcc is with coverage support"
+    else
+	echo "[-] gcc is not with coverage support"
+	exit 1
+    fi
+}
+
+function patch_linux_kernel {
+    echo "[*] patch linux kernel"
+    if [[ ! -f "$INIT_DIR/$KERNEL_SOURCE_FILE" ]]; then
+	echo "[*] DOWNLOAD $KERNEL_SOURCE_FILE"
+	wget -O "$INIT_DIR/$KERNEL_SOURCE_FILE" -c "$KERNEL_SOURCE_URL"
+    else
+	echo "[+] $KERNEL_SOURCE_FILE FOUND"
+    fi
+
+    if [[ ! -f "$INIT_DIR/$KERNEL_PATCH_FILE" ]]; then
+	echo "[*] DOWNLOAD $KERNEL_PATCH_FILE"
+	wget -O "$INIT_DIR/$KERNEL_PATCH_FILE" -c "$KERNEL_PATCH_URL"
+    else
+	echo "[+] $KERNEL_PATCH_FILE FOUND"
+    fi
+
+    if [[ ! -d "$KERNEL_SOURCE_DIR" ]]; then
+        tar xvf "$KERNEL_SOURCE_FILE" >/dev/null
+        cd "$KERNEL_SOURCE_DIR"
+        xz -cd ../"$KERNEL_PATCH_FILE" | patch -R -p1
+    fi
+}
+
+function config_with_sed {
+    sed -i "s/^# $1 .*/$1=y/" "$KCGF_FILE"
```

Comment:
The proper way to do this is with `scripts/kconfig/merge_config.sh` in kernel tree.